src/edu/cmu/tartan/Game.java

```diff
@@ -564,9 +611,200 @@ private void help() {
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        printOutInterface.console("Save Done");
+	}
+
+	public List<Room> loadRoomArrayList(JSONObject roomList){
+        //make load rooms status
+        List<Room> loadedRooms = new ArrayList<>();
+
+        for(int i = 0; i<roomList.size(); i++){
+
+            JSONObject roomNumber = (JSONObject) roomList.get("room"+i);
+
+            //get room description and make room
+            String roomDescription = (String) roomNumber.get("room");
+            Room room = new Room("",roomDescription);
+
+            //make item list
+            JSONArray itemList = (JSONArray) roomNumber.get("items");
+            ArrayList<Item> itemArrayList = new ArrayList<>();
+            Iterator<String> iterator = itemList.iterator();
+            while (iterator.hasNext()){
+                itemArrayList.add(Item.getInstance(iterator.next()));
+            }
+
+            //clear item list then put new item list
+            room.items.clear();
+            room.putItems(itemArrayList);
+
+            loadedRooms.add(room);
+        }
+
+        return loadedRooms;
+    }
+
+    public void updateRoomsItemList(List<Room> loadedRooms){
+        for(Room gameRoom : roomArrayList){
+            for(Room loadRoom : loadedRooms){
+                if(gameRoom.shortDescription().equals(loadRoom.shortDescription())){
+                    gameRoom.items.clear();
+                    gameRoom.putItems(loadRoom.items);
+                }
+            }
+        }
+    }
+
+    public void updateCurrentRoom(String currentRoom){
+        //set current room
+
+        for(Room gameRoom : roomArrayList){
+            if(gameRoom.toString().compareTo(currentRoom) == 0){
+                player.setCurrentRoom(gameRoom);
+                break;
+            }
+        }
+    }
+
+    public void updateVisitedRoomList(JSONArray visitiedRooms){
+        Iterator<String> iterator = visitiedRooms.iterator();
+        if(iterator.hasNext()){
+            //clear visited room list
+            player.getRoomsVisited().clear();
+        }
+        while (iterator.hasNext()){
+            String visitedRoom = iterator.next();
+            for(Room gameRoom :roomArrayList){
+                if(gameRoom.shortDescription().compareTo(visitedRoom) == 0){
+                    gameRoom.setRoomWasVisited(true);
+                    player.getRoomsVisited().add(gameRoom);
+                }
+            }
+        }
+    }
+
+    public void updateCollectedItemList(JSONArray collectedItems){
+        Iterator<String> collectedItemsiterator = collectedItems.iterator();
+
+        //clear player's collected items
+        player.getCollectedItems().clear();
+
+        while (collectedItemsiterator.hasNext()){
+            player.grabItem(Item.getInstance(collectedItemsiterator.next()));
+        }
+    }
+
+	public synchronized void load(){
```

Comment:
If the exception or some problem happens during load(), next operation after the exception doesn't work.